groups/ntc/ntcr/ntcr_streamsocket.cpp

```diff
@@ -3652,6 +3652,42 @@ ntsa::Error StreamSocket::privateOpen(
         return error;
     }
 
+    if (d_options.timestampOutgoingData().has_value()) {
+        d_timestampOutgoingData = d_options.timestampOutgoingData().value();
+    }
+
+    if (d_timestampOutgoingData) {
+        ntsa::SocketOption option;
+        option.makeTimestampOutgoingData(true);
+
+        error = streamSocket->setOption(option);
+        if (error) {
+            BSLS_LOG_DEBUG("Failed to set socket option: "
+                           "timestamp outcoming data: %s",
+                           error.text().c_str());
+
+            d_timestampOutgoingData = false;
+        }
+    }
+
+    if (d_options.timestampIncomingData().has_value()) {
+        d_timestampIncomingData = d_options.timestampIncomingData().value();
+    }
+
+    if (d_timestampIncomingData) {
+        ntsa::SocketOption option;
+        option.makeTimestampIncomingData(true);
+
+        error = streamSocket->setOption(option);
+        if (error) {
+            BSLS_LOG_DEBUG("Failed to set socket option: "
```

Comment:
Shouldn't we use `NTCI_LOG_DEBUG` everywhere instead of `BSLS_LOG_DEBUG`?
And maybe warning would suit better for this case?

Reply:
Done, but I'm keeping the DEBUG severity level. WARN is too loud for a requested-but-unavailable feature, not required for operation.